keps/sig-auth/2862-fine-grained-kubelet-authz/README.md

```diff
@@ -0,0 +1,1046 @@
+[kubernetes/kubernetes]: https://git.k8s.io/kubernetes
+[kubernetes/website]: https://git.k8s.io/website
+
+## Summary
+
+<!--
+This section is incredibly important for producing high-quality, user-focused
+documentation such as release notes or a development roadmap. It should be
+possible to collect this information before implementation begins, in order to
+avoid requiring implementors to split their attention between writing release
+notes and implementing the feature itself. KEP editors and SIG Docs
+should help to ensure that the tone and content of the `Summary` section is
+useful for a wide audience.
+
+A good summary is probably at least a paragraph in length.
+
+Both in this section and below, follow the guidelines of the [documentation
+style guide]. In particular, wrap lines to a reasonable length, to make it
+easier for reviewers to cite specific portions, and to minimize diff churn on
+updates.
+
+[documentation style guide]: https://github.com/kubernetes/community/blob/master/contributors/guide/style-guide.md
+-->
+
+We propose a change to the Kubelet API authorization to provide more
+fine-grained control. Currently, the API uses a coarse authorization scheme,
+where actions like reading health status and the ability to exec into a pod
+require the same permission.
+
+## Motivation
+
+<!--
+This section is for explicitly listing the motivation, goals, and non-goals of
+this KEP.  Describe why the change is important and the benefits to users. The
+motivation section can optionally provide links to [experience reports] to
+demonstrate the interest in a KEP within the wider Kubernetes community.
+
+[experience reports]: https://github.com/golang/go/wiki/ExperienceReports
+-->
+
+Historically the `/healthz` and 
+`/pods` endpoints were available on the unauthenticated kubelet read-only port(10255) 
+but kubelet read-only port is disabled by default and enabling the port is
+considered a security worst practice. This has caused a lot of monitoring and 
+logging agents to switch to using the kubelet authenticated port (10250).
+However, the Kubelet API on the authenticated(10250) port currently uses a very
+coarse scheme for authorizing requests. For example, reading `/healthz` and 
+calling `/exec/…` (i.e. execute arbitrary code) require the same `proxy` 
+subresource. So, if an application needs to say `list` `pods` on the node, as 
+many monitoring and logging agents do, they must be granted the `proxy` 
+subresource in RBAC. Doing so grants many other powerful permissions to these 
+agents which could be exploited by an attacker to escalate privilege.  
+
+### Goals
+
+<!--
+List the specific goals of the KEP. What is it trying to achieve? How will we
+know that this has succeeded?
+-->
+- Introduce new authorization subresources for `configz`, `/healthz` and `/pods/` kubelet endpoints,
+allowing for more  granular authorization.
+- Officially document the `configz`, `healthz` and `/pods/` endpoints.
```

Comment:
Aside: we should document the kubelet anyway; you don't need a KEP to document existing stable features.

Reply:
In sig node we discussed making this a part of the goals of this KEP. These endpoints were previously undocumented and they suggested that we document them if we are going to add fine-grained authorization for these endpoints. So I'll leave them as a goal of this KEP.